solidity/contracts/cross-chain/wormhole/L1BTCDepositorNttWithExecutor.sol

```diff
@@ -0,0 +1,1102 @@
+            amount
+        );
+
+        // Execute the transfer with executor support
+        uint64 sequence = nttManagerWithExecutor.transfer{value: msg.value}( // slither-disable-line reentrancy-vulnerabilities-3
+            underlyingNttManager,
+            amount,
+            destinationChain,
+            actualRecipient,
+            bytes32(uint256(uint160(msg.sender))), // refundAddress as bytes32
+            "", // Empty transceiver instructions for basic transfer
+            executorArgs,
+            feeArgs
+        );
+
+        emit TokensTransferredNttWithExecutor( // slither-disable-line reentrancy-vulnerabilities-3
+            msg.sender,
+            nonce,
+            amount,
+            destinationChain,
+            actualRecipient,
+            sequence,
+            destinationChainReceiver,
+            msg.value
+        );
+    }
+
+    /// @notice Extract destination chain from encoded receiver address
+    /// @param destinationChainReceiver Encoded receiver with chain ID in first 2 bytes
+    /// @return chainId The destination chain ID
+    function _getDestinationChainFromReceiver(bytes32 destinationChainReceiver)
+        internal
+        view
+        returns (uint16 chainId)
+    {
+        chainId = uint16(bytes2(destinationChainReceiver));
+
+        // CRITICAL: No fallback to default chain - user must specify valid chain
+        if (chainId == 0) {
+            revert("Chain ID cannot be zero");
+        }
+
+        if (!supportedChains[chainId]) {
+            revert("Destination chain not supported");
+        }
+
+        return chainId;
+    }
+
+    /// @notice Get the default supported chain ID
+    /// @return chainId The default supported chain ID
+    function _getDefaultSupportedChain()
+        internal
+        view
+        returns (uint16 chainId)
+    {
+        return defaultSupportedChain;
+    }
+
+    /// @notice Extract recipient address from encoded receiver data
+    /// @param destinationChainReceiver Encoded receiver data
+    /// @return recipient The recipient address (last 30 bytes, padded to 32 bytes)
+    function _getRecipientAddressFromReceiver(bytes32 destinationChainReceiver)
+        internal
+        pure
+        returns (bytes32 recipient)
+    {
+        return
+            bytes32(
+                uint256(destinationChainReceiver) &
+                    0x0000FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF
+            );
+    }
+
+    /// @notice Validates the format of a signed quote from Wormhole Executor API
+    /// @param signedQuote The signed quote bytes to validate
+    /// @dev Keep validation minimal - NttManagerWithExecutor handles detailed validation
+    function _validateSignedQuoteFormat(bytes memory signedQuote)
+        internal
+        pure
+    {
+        require(signedQuote.length > 0, "Signed quote cannot be empty");
+        require(signedQuote.length >= 32, "Signed quote too short");
```

Comment:
Are we checking the actual signature on the quote anywhere?

Reply:
The quote may also become stale, as we discussed offline. Are we checking for quote freshness anywhere?